x-pack/plugins/alerting/server/index.ts

```diff
@@ -79,8 +78,11 @@ export const plugin = async (initContext: PluginInitializerContext) => {
   return new AlertingPlugin(initContext);
 };
 
-export const config: PluginConfigDescriptor<AlertsConfigType> = {
+export const config: PluginConfigDescriptor<AlertingConfig> = {
   schema: configSchema,
+  exposeToBrowser: {
```

Comment:
If we use `exposeToBrowser`, then in theory we don't need this added to the triggers_actions_ui config route.  Though `exposeToBrowser` only makes it available to the browsers alerting plugin, not t_a_ui - though obviously we could make it accessible.

But I don't think we need both, so we should pick one and not do the other.  We don't need two ways to do the same thing ...

Or perhaps I missed something ...

Reply:
I'm not sure if I understand, the only `triggers_actions_ui` code that has been modified here was some test files to align with the mock alerting plugin. But to your larger point, I agree - the `exposeToBrowser` has been implemented here and other plugins can use them if they so choose. That's how we're utilizing it in security solution

Reply:
The way this is structured we are also changing the output of the HTTP endpoint `/internal/triggers_actions_ui/_config` - to return ALL the values under `run` ,not just the alerts.max value -
which is something we need to consider re: backwards compatibility, documentation, and security (should we be exposing these via API).  Do we really need to change the output of this endpoint?

```console
$ curl $KB_URL/internal/triggers_actions_ui/_config
{"minimumScheduleInterval":{"value":"1s","enforce":false},"maxScheduledPerMinute":10000,"run":{"actions":{"max":100000},"alerts":{"max":1000}},"isUsingSecurity":true}
```

I believe it's also the case that the values returned by this endpoint are or can be calculated, so using the value from the config wouldn't work.  Which is why we have this endpoint.  Static values (I assume these are static) can just use the `exposeToBrowser` path.

Reply:
In the end, I would like to NOT change the output of the `_config` route (that's adding the `run` props) - not so much because it's a security concern now, but feels like a slippery slope to having a problem later, if someone follows this pattern, and we do leak something we shouldn't.

Feels like we need to remove `run` from `AlertingRulesConfig`, or change the `_config` route in t_a_ui to `pick` the fields it should be returning from the bigger config object (the existing ones, but not `run`).

Reply:
I think the second way is probably the more preferable method, just so we can use that `getConfig` function elsewhere in other plugins without changing the `_config` route output. Right now we're using the `getConfig` function to compare on the server side in security solution as well which is why the `run` props were added in the first place. If y'all are ok with exposing the config values under `run` to that internal `getConfig` method from the plugin setup object and modifying the `triggers_actions_ui` config route so that we're locked into the existing values, I can change that over.

Reply:
I don't have a problem exposting this at the plugin level - my concern is at the HTTP response level.

Filtering **IN** just the props we want returned from that `_config` endpoint would be perfect, as it means we won't have to worry about accidently leaking things later.  So, in theory, `x-pack/plugins/triggers_actions_ui/server/routes/config.test.ts` won't have any changes, but presumably it's pair `config.ts` will.

Reply:
Ok that all sounds good, I think we're on the same page 👍